pkg/controller/sharedcontrollerfactory.go

```diff
@@ -23,11 +23,22 @@ type SharedControllerFactory interface {
 }
 
 type SharedControllerFactoryOptions struct {
+	CacheOptions *cache.SharedCacheFactoryOptions
+
 	DefaultRateLimiter workqueue.RateLimiter
 	DefaultWorkers     int
 
 	KindRateLimiter map[schema.GroupVersionKind]workqueue.RateLimiter
 	KindWorkers     map[schema.GroupVersionKind]int
+
+	// SkipNoChangeUpdate causes the handle function to only proceed if the object was actually updated.
+	// This is intended to be used by  applications with many objects and/or controllers types that
+	// have alternative means of rerunning when necessary. When the informer's resync their cache the
+	// update function is run. This reducing the overhead but has the tradeoff of not rerunning handlers
+	// that may rely on external objects or services or experience a bug causing them to not retry even
+	// if it was necessary. The result is that running the handler func on resync will mostly only serve
+	// the purpose of catching missed cache events.
+	SkipNoChangeUpdate bool
```

Comment:
Can we rename this to something like NoopResync or something like that? SkipNoChange is a bit of a confusing double negative - it's not doing something if its not a change?

Making the name consistent across both structs would also be nice.

Reply:
I feel like the name Noop Resync would be inaccurate since there is an op if the resource versions happen to be different. I also don't think it's a double negative since that is where two negatives achieve the same meaning as a positive which is not what's happening here. The name means that we are skipping the handle function if there is no change in the objects. We can change it to HandleNoChangeUpdate but I'd prefer if the default bool value, false, keeps this feature disabled. Maybe `OnlyUpdateOnObjectChange`?

Reply:
How about `SyncOnlyChangedObjects`